File: app/src/main/java/xyz/michaelobi/paperplayer/widget/PlayPauseButton.java
package xyz.michaelobi.paperplayer.widget;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;

import xyz.michaelobi.paperplayer.R;

/**
 * TODO: document your custom view class.
 */
public final class PlayPauseButton extends android.support.v7.widget.AppCompatImageButton {
    private final Context context;

    public PlayPauseButton(Context context) {
        super(context);
        this.context = context;
    }

    public PlayPauseButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
    }

    public PlayPauseButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        this.context = context;
    }

    public void play() {
        changeState(ContextCompat.getDrawable(context, R.drawable.ic_play_arrow_white_24dp));
    }

    public void pause() {
        changeState(ContextCompat.getDrawable(context, R.drawable.ic_pause_white_24dp));
    }

    private void changeState(Drawable drawable) {
        setImageDrawable(drawable);
    }
}

File: app/src/main/java/xyz/michaelobi/paperplayer/widget/PlayPauseButton.kt
/*
 * MIT License
 *
 * Copyright (c) 2017 MIchael Obi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package xyz.michaelobi.paperplayer.widget

import android.content.Context
import android.graphics.drawable.Drawable
import android.support.v4.content.ContextCompat
import android.support.v7.widget.AppCompatImageButton
import android.util.AttributeSet
import xyz.michaelobi.paperplayer.R

/**
 * An [AppCompatImageButton] with two playback states for media players:

 *
 * play() and pause() methods are called to alter these states updating the Image [ ]
 */
class PlayPauseButton : AppCompatImageButton {

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    fun play() {
        changeState(ContextCompat.getDrawable(context, R.drawable.ic_play_arrow_white_24dp))
    }

    fun pause() {
        changeState(ContextCompat.getDrawable(context, R.drawable.ic_pause_white_24dp))
    }

    private fun changeState(drawable: Drawable) {
        setImageDrawable(drawable)
    }
}